pmm-app/src/pmm-settings/components/Advanced/Advanced.tsx

```diff
@@ -76,75 +68,90 @@ export const Advanced: FC<AdvancedProps> = ({
 
   return (
     <div className={styles.advancedWrapper}>
-      <table>
-        <tbody>
-          <tr>
-            <td>
-              <div
-                className={settingsStyles.labelWrapper}
-                data-qa="advanced-label"
-              >
-                <span>{retentionLabel}</span>
-                <LinkTooltip
-                  tooltipText={retentionTooltip}
-                  link={DATA_RETENTION_URL}
-                  linkText={tooltipLinkText}
-                  icon="info-circle"
+      <Form
+        onSubmit={applyChanges}
+        initialValues={initialValues}
+        render={({
+          values, handleSubmit, valid, pristine
+        }) => (
+          <form onSubmit={handleSubmit}>
+            <table>
+              <tbody>
+                <tr>
+                  <td>
+                    <div
+                      className={settingsStyles.labelWrapper}
+                      data-qa="advanced-label"
+                    >
+                      <span>{retentionLabel}</span>
+                      <LinkTooltip
+                        tooltipText={retentionTooltip}
+                        link={DATA_RETENTION_URL}
+                        linkText={tooltipLinkText}
+                        icon="info-circle"
+                      />
+                    </div>
+                  </td>
+                  <td className={styles.retentionInputWrapper}>
+                    <Field
```

Comment:
Maybe we should think about moving SwitchRow and NumericInputField to shared/components instead of using Field here? It will simplify the code and will help us to hide unneeded things inside

@tiagomotasantos  @atymchuk @nicolalamacchia what do you think?

Reply:
NumericInputField is on the shared folder already. SwitchRow was a bit specific to this that's why I didn't move it, but yeah it's up to discussion.